src/system_helper/filesystem_helper.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * SPDX-FileCopyrightText: 2023 UnionTech Software Technology Co., Ltd.
+ *
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+#include "filesystem_helper.h"
+
+#include "module/dbus_ipc/dbus_common.h"
+#include "module/util/runner.h"
+#include "privilege/privilege_install_portal.h"
+
+#include <QDir>
+
+#include <sys/stat.h>
+
+namespace linglong {
+namespace system {
+namespace helper {
+
+bool isWritable(const char *path, const char *procPath)
+{
+    struct stat buf = {};
+
+    stat(procPath, &buf);
+    uid_t procUid = buf.st_uid;
+    gid_t procGid = buf.st_gid;
+
+    stat(path, &buf);
+    if (procUid == buf.st_uid && buf.st_mode & S_IWUSR) {
+        return true;
+    } else if (procGid == buf.st_gid && buf.st_mode & S_IWGRP) {
+        return true;
+    } else if (buf.st_mode & S_IWOTH) {
+        return true;
+    }
+    return false;
+}
+
+/*!
+ * mount file to sub path of "/run/user/{uid}/linglong"
+ * @param mountPoint
+ * @param options
+ */
+void FilesystemHelper::Mount(const QString &source,
+                             const QString &mountPoint,
```

Comment:
I think we should make mountPoint always be under `/run/user/{uid}/linglong`.

Should we make the mountPoint argument to be a relative path from `/run/user/{uid}/linglong`?